tests/state.test.js

```diff
@@ -158,6 +158,35 @@ exports['given a state machine'] = {
     assert(machine._guard)
     machine.close()
     assert(!machine._guard)
+  },
+
```

Comment:
👍 for testing the 3 different scenarios (attaching handler, overriding handler, handler is function only)

Btw you can harden your tests by asserting on function reference, e.g.:
```
// Attaching
var handler = function() {};

machine.attachErrorHandler(handler);
assert.equal(machine.errorHandler, handler);
```
```
// Overriding
var handler1 = function() {};
var handler2 = function() {};

machine.attachErrorHandler(handler1);
machine.attachErrorHandler(handler2);
assert.equal(machine.errorHandler, handler2);
```